tldraw/packages/core/src/lib/TLApi/TLApi.ts

```diff
@@ -161,4 +161,16 @@ export class TLApi<S extends TLShape = TLShape, K extends TLEventMap = TLEventMa
     this.app.save()
     return this
   }
+
+  undo = () => {
```

Comment:
I think there is an issue related to when/how `history.persisted` is called and that's why we need to undo/redo multiple times programatically.